packages/models-library/src/models_library/api_schemas_storage/storage_schemas.py

```diff
@@ -46,6 +50,12 @@
 DatCoreDatasetName: TypeAlias = Annotated[
     str, StringConstraints(pattern=DATCORE_DATASET_NAME_RE)
 ]
+DatCoreCollectionName: TypeAlias = Annotated[
+    str, StringConstraints(pattern=DATCORE_COLLECTION_NAME_RE)
```

Comment:
i really like this way much more that using old classes that inherited from str but were not str!